SmartDeviceLinkTests/ProxySpecs/SDLHapticManagerSpec.m

```diff
@@ -0,0 +1,169 @@
+//
+//  SDLHapticManagerSpec.m
+//  SmartDeviceLink-iOS
+//
+//  Copyright © 2017 smartdevicelink. All rights reserved.
+//
+
+#import <Quick/Quick.h>
+#import <Nimble/Nimble.h>
+#import <OCMock/OCMock.h>
+
+#import "SDLHapticManager.h"
+#import "SDLSendHapticData.h"
+#import "SDLManager.h"
+#import "SDLTouchCoord.h"
+#import "SDLTouchEvent.h"
+#import "SDLTouch.h"
+
+SDLTouch* generateTouchEvent(int xCoord, int yCoord)
+{
+    unsigned long timeStamp = 10;
+    SDLTouchCoord* firstCoord = [[SDLTouchCoord alloc] init];
+    firstCoord.x = [NSNumber numberWithInt:xCoord];
+    firstCoord.y = [NSNumber numberWithInt:yCoord];
+    
+    SDLTouchEvent* firstTouchEvent = [[SDLTouchEvent alloc] init];
+    firstTouchEvent.touchEventId = @0;
+    firstTouchEvent.coord = [NSMutableArray arrayWithObject:firstCoord];
+    firstTouchEvent.timeStamp = [NSMutableArray arrayWithObject:@(timeStamp)];
+    SDLTouch* firstTouch = [[SDLTouch alloc] initWithTouchEvent:firstTouchEvent];
+    return firstTouch;
+}
+
+QuickSpecBegin(SDLHapticManagerSpec)
+
+describe(@"the haptic manager", ^{
+    
+    __block UIWindow *uiWindow;
+    __block UIViewController *uiViewController;
+    
+    __block SDLHapticManager *hapticManager;
+    __block SDLSendHapticData* capturedHapticRPC;
+    
+    __block id sdlManagerMock = OCMClassMock([SDLManager class]);
+    
+    beforeEach(^{
+        uiWindow = [[UIWindow alloc] init];
+        uiViewController = [UIViewController alloc];
+        
+        [uiWindow addSubview:uiViewController.view];
+        
+        OCMExpect( [[sdlManagerMock stub] sendRequest:[OCMArg checkWithBlock:^BOOL(id value){
+            BOOL isFirstArg = [value isKindOfClass:[SDLSendHapticData class]];
+            if(isFirstArg)
+            {
+                capturedHapticRPC = value;
+            }
+            return YES;
+        }] withResponseHandler:[OCMArg any]]);
+
+    });
+    
+    context(@"when initialized with no view", ^{
+        beforeEach(^{
+            hapticManager = [[SDLHapticManager alloc] initWithWindow:uiWindow sdlManager:sdlManagerMock];
+        });
+        
+        it(@"should have no view", ^{
+            OCMVerify(sdlManagerMock);
+            expect([[capturedHapticRPC hapticRectData] count]).to(equal(0));
+        });
+    });
+    
+    context(@"when initialized with single view", ^{
+        beforeEach(^{
+            UITextField *textField1 = [[UITextField alloc]  initWithFrame:CGRectMake(101, 101, 50, 50)];
```

Comment:
There should also be a test to make sure that the captured coordinates are correct.